src/wasm-binary.h

```diff
@@ -357,6 +355,12 @@ enum EncodedType {
   Empty = -0x40 // 0x40
 };
 
+enum EncodedHeapType {
+  func = -0x10,    // 0x70
+  extern_ = -0x11, // 0x6f
+  exn = -0x18,     // 0x68
+};
```

Comment:
While these values currently match the ones of the respective `ref` in `EncodedType`, that's not the case for those we'll need to add later. For instance, soon there'll also be

```
  i31 = -0x17,       // 0x69
```

which doesn't match `i31ref` (`-0x16`) and

```
  signature = -0x20, // 0x60
  struct_ = -0x21,   // 0x5f	
  array = -0x22      // 0x5e
```

which don't have a corresponding type, so I think this makes sense given that reference types already introduce the concept.

Reply:
Sounds good. Do you have an intuition for why i31 uses a different encoding than i31ref?

Reply:
Perhaps because of the nullability difference, but not sure. Can be seen here: https://github.com/WebAssembly/gc/blob/master/proposals/gc/MVP.md#reference-types-1, with `-0x17` in types indicating an `(rtt n ht)`.